llvm/lib/CodeGen/DeadMachineInstructionElim.cpp

```diff
@@ -157,6 +157,15 @@ bool DeadMachineInstructionElimImpl::runImpl(MachineFunction &MF,
 bool DeadMachineInstructionElimImpl::eliminateDeadMI(MachineFunction &MF) {
   bool AnyChanges = false;
 
+  // Cache the reserved registers that can be simplified.
+  BitVector ReservedRegs = MRI->getReservedRegs();
+  SmallVector<MCPhysReg, 16> SimplifiableReservedRegs;
+  for (MCPhysReg PhysReg : ReservedRegs.set_bits()) {
+    if (MRI->canSimplifyPhysReg(PhysReg)) {
+      SimplifiableReservedRegs.push_back(PhysReg);
+    }
+  }
```

Comment:
Nit: outline in something like `SmallVector<MCPhysReg, 16> getSimplifiableReservedRegs(const MachineRegisterInfo &MRI);`

Reply:
done.